src/main/scala/dpla/ingestion3/CdlHarvesterMain.scala

```diff
@@ -0,0 +1,176 @@
+package dpla.ingestion3
+
+import java.io.File
+
+import dpla.ingestion3.harvesters.api._
+import dpla.ingestion3.harvesters.file.NaraFileHarvestMain.getAvroWriter
+import dpla.ingestion3.utils.{FlatFileIO, Utils}
+import org.apache.avro.Schema
+import org.apache.avro.file.DataFileWriter
+import org.apache.avro.generic.{GenericData, GenericRecord}
+import org.apache.log4j.LogManager
+import org.json4s.JsonAST.JInt
+import org.json4s.jackson.JsonMethods._
+import org.rogach.scallop.{ScallopConf, ScallopOption}
+
+
+object CdlHarvesterMain {
+
+  private val logger = LogManager.getLogger(getClass)
+
+  /**
+    * Driver for harvesting from California Digital Library's Solr endpoint
+    * (https://solr.calisphere.org/solr/query).
+    *
+    * Expects command line args of
+    * --endpoint=solr.calisphere.org.
+    * --apiKey - The CDL api key
+    * --rows - Number of records to return per page. Defaults to 10
+    * --outputFile - Location to save the avro. Defaults to ./out
+    * --query - Solr query to select the records. Defaults to *:*
+    *
+    * @param args
+    */
+  def main(args: Array[String]): Unit = {
+    // Parse and set the command line options
+    val conf = new CdlHarvestConf(args)
+    val outFile = new File(conf.outputFile.getOrElse("out"))
```

Comment:
I cribbed this from the NARA harvester but it will drop it inside the current working project directory and the recursive delete could wreck havoc here (I've deleted my project directory more than once)